packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -507,17 +510,41 @@ function voidFunctionParams(
       : subType.getConstructSignatures();
     for (const signature of signatures) {
       for (const [index, parameter] of signature.parameters.entries()) {
-        const type = checker.getTypeOfSymbolAtLocation(
+        const decl = parameter.getDeclarations()?.[0];
+        let type = checker.getTypeOfSymbolAtLocation(
           parameter,
           node.expression,
         );
+
+        const indices = [index];
+        // If this is a array 'rest' parameter, add all of the remaining parameter indices.
+        // Note - we currently do not support 'spread' arguments
```

Comment:
Could you file an issue to support spread arguments, and link it in the code comment please? It's easy to lose track of these things, but +1 to supporting them eventually.

Reply:
Opened https://github.com/typescript-eslint/typescript-eslint/issues/5744